x-pack/plugins/security_solution/public/common/utils/exception_list_items/mappers.ts

```diff
@@ -0,0 +1,218 @@
+  switch (hash.length) {
+    case 32:
+      return 'md5';
+    case 40:
+      return 'sha1';
+    case 64:
+      return 'sha256';
+  }
+};
+
+const createEntryMatch = (field: string, value: string): EntryMatch => {
+  return { field, value, type: 'match', operator: OPERATOR_VALUE };
+};
+
+const createEntryMatchAny = (field: string, value: string[]): EntryMatchAny => {
+  return { field, value, type: 'match_any', operator: OPERATOR_VALUE };
+};
+
+const createEntryMatchWildcard = (field: string, value: string): EntryMatchWildcard => {
+  return { field, value, type: 'wildcard', operator: OPERATOR_VALUE };
+};
+
+const createEntryNested = (field: string, entries: NestedEntriesArray): EntryNested => {
+  return { field, entries, type: 'nested' };
+};
+
+function groupHashEntry(conditionEntry: ConditionEntry): EntriesArray {
+  const entriesArray: EntriesArray = [];
+
+  if (!Array.isArray(conditionEntry.value)) {
+    const entry = createEntryMatch(
+      `process.hash.${hashType(conditionEntry.value)}`,
+      conditionEntry.value.toLowerCase()
+    );
+    entriesArray.push(entry);
+    return entriesArray;
+  }
+
+  const hashTypeGroups: { md5: string[]; sha1: string[]; sha256: string[] } =
+    conditionEntry.value.reduce(
+      (memo, val) => {
+        const type = hashType(val);
+        if (!type) return memo;
+
+        return {
+          ...memo,
+          [type]: [...memo[type], val],
+        };
+      },
+      {
+        md5: [],
+        sha1: [],
+        sha256: [],
+      } as { md5: string[]; sha1: string[]; sha256: string[] }
+    );
+  Object.entries(hashTypeGroups).forEach(([type, values]) => {
+    if (!values.length) {
+      return;
+    }
+
+    const entry = createEntryMatchAny(`process.hash.${type}`, values);
+    entriesArray.push(entry);
+  });
+
+  return entriesArray;
+}
+
+function createNestedSignatureEntry(
+  value: string | string[],
+  isTrustedApp: boolean = false
+): EntryNested {
+  const subjectNameMatch = Array.isArray(value)
+    ? createEntryMatchAny('subject_name', value)
+    : createEntryMatch('subject_name', value);
+  const nestedEntries: EntryNested['entries'] = [];
+  if (isTrustedApp) nestedEntries.push(createEntryMatch('trusted', 'true'));
+  nestedEntries.push(subjectNameMatch);
+  return createEntryNested('process.Ext.code_signature', nestedEntries);
+}
+
+function createWildcardPathEntry(value: string | string[]): EntryMatchWildcard | EntryMatchAny {
+  return Array.isArray(value)
+    ? createEntryMatchAny('process.executable.caseless', value)
+    : createEntryMatchWildcard('process.executable.caseless', value);
+}
+
+function createPathEntry(value: string | string[]): EntryMatch | EntryMatchAny {
+  return Array.isArray(value)
+    ? createEntryMatchAny('process.executable.caseless', value)
+    : createEntryMatch('process.executable.caseless', value);
+}
+
+export const conditionEntriesToEntries = (
+  conditionEntries: ConditionEntry[],
+  isTrustedApp: boolean = false
```

Comment:
considered making this more extensible such as passing entry overrides or something but figured we shouldn't over-engineer it right now.